### ISSUE TITLE
go.twitch.tv beta website support

### ISSUE THREAD

**Issue by libix** [FrankerFaceZ/FrankerFaceZ#241]
**Web Browser**: Chrome/60.0.3112.113 

**Do you use BetterTTV or other Twitch extensions**: BTTV and FFZ 

**Bug / Idea**: the extension don't work with the new beta website, is it intended until things are pretty stable twitch side ? or is it planned for later ?

**Comment by SirStendec** [comment]
I am currently working on support for the beta website, but despite looking similar, the website is entirely different behind the scenes and I have to rewrite the entirety of FrankerFaceZ.

(Leaving this issue open so people see it and don't open more.)

**Comment by sparcut** [comment]
For the better though, React > Ember

**Comment by SirStendec** [comment]
No, not for better. React is awful. It does not trust browsers to do their jobs. It reimplements things that already exist, like the event system. What's wrong with the event system? Nothing. But Facebook wanted to make their own.

React is a mess of licensing because of Facebook's patents.

The one thing React is supposed to be is fast, but in my performance testing, chat causes frame drops as often as it did with Ember. The one thing it has accomplished at this point is faster load times, but I consider that a load of bunk because they are using webpack now and making the site modular, which they could have done just as well with Ember. They also could have used SSR with Ember. Not to mention the beta site isn't nearly feature complete compared to the Ember app.

The only thing react has actually changed so far is that it has made the website a complete and utter pain in the ass to work with. With Ember, I can reopen components and modify their behavior. I get all the access I could need to the data. I can easily tweak native behaviors.

With React, I can *kind of* find instances of things by finding a DOM node and then looping over its attributes for something that starts with `__reactInstance$`. Good luck modifying any logic or anything. Pretty much have to make changes to the DOM to make things happen.

And React can decide randomly to just revert the DOM to what it thinks the DOM should be.

Basically, the next versions of BTTV and FFZ are probably going to be a mess and not work as well as our users expect, and it's the fault of whichever boneheaded developer clique inside of Twitch decided to force React on everyone instead of picking a decent framework or, better, just fixing the problem with the Ember website. All they needed was modularity, maybe SSR, and a CSS refactor to make chat reflows less expensive.

I am the least happy I could be with React. I know Twitch developers who are not happy with their move to React. React is a mess. This is not a change for the better.

**Comment by sparcut** [comment]
@SirStendec I can understand the pain regarding modification of live DOM and components, but whenever I've seen Ember used for large-scale applications, they've always seemed sluggish at times.

Be that the times I've looked under the hood and found Ember mightn't be representative of Ember sites as a whole, only those that I've looked at the code trying to figure out why it's sluggish.

I'd rather not turn this into a React v Ember debate, but I can totally see where the flaws since React can decide to do whatever it feels like when it wants. If Twitch are going to fully move to React, they'll **have to** add some sort of support or API you can grab onto, since, as soon as it bubbles up to the average user that it's Twitch's fault BTTV and FFZ no longer work properly, they'll already be knee deep in shit.

**Comment by gempir** [comment]
The thing is twitch doesn't want people to develop extension like FFZ/BTTV it seems. They want developers only using APIs they openly document like Extensions, IRC, etc.

**Comment by Corrodias** [comment]
If that is the case... Maybe they should make said api halfway capable so
that ffz *could be* reimplemented with it, before messing everything up.
That would be a reasonable development practice.

On Oct 3, 2017 11:34 AM, "Daniel Pasch" <notifications@github.com> wrote:

The thing is twitch doesn't want people to develop extension like FFZ/BTTV
it seems. They want developers only using APIs they openly document like
Extensions, IRC, etc.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/FrankerFaceZ/FrankerFaceZ/issues/241#issuecomment-333903144>,
or mute the thread

**Comment by fmauNeko** [comment]
@SirStendec they removed the patent thingy from v16

**Comment by serg06** [comment]
Temporary fix: Click the user menu in top-right and tick "Legacy Site".

**Comment by Chronophylos** [comment]
@serg06 that doesn't work anymore. The legacy site is not available anymore.
